### PR TITLE
Traffic change waits the stack to be ready

### LINKED ISSUE #398: Senza traffic should wait for stack to be ready to change traffic
### Current Behaviour

When running change traffic `senza traffic` on stack recently deployed. The status code of the stack is not ready (`(CRETE|UPDATE)_COMPLETE`) and Senza fails trying to update it.
### Cause of the Problem

It is not possible to update CF stack when it's status code is not in `(CRETE|UPDATE)_COMPLETE` (or we should not change when it is not in one of those status code).
### Expected Behaviour

Senza should wait the stack to get in a final status code if the current status of the stack is `(CREATE|UPDATE)_IN_PROGRESS` and ending up in some good state `(CREATE|UPDATE)_COMPLETE`.

### PR DESCRIPTION
**Pull request by rafaelcaricio**
Fixes #398

### CODE REVIEW

File: senza/cli.py [hunk 1]
```diff
@@ -1152,7 +1152,26 @@ def traffic(stack_name, stack_version, percentage, region, output):
...
+                        current_stack_status = stack_to_change.StackStatus
+
+                        if current_stack_status.endswith('_COMPLETE'):
+                            change_version_traffic(ref, percentage, region)
+                            break
+                        elif current_stack_status.endswith('_IN_PROGRESS'):
+                            # some operation in progress, let's wait some time to try again
+                            time.sleep(5)
+                        else:
+                            error("Stack is not in a valid state (%s), traffic change cannot be performed.",
```
Comment (hjacobs) [inline senza/cli.py]: the message format looks wrong, I'm pretty sure it will not work that way (?)

**Comment by rafaelcaricio** [comment]
You are right. I have to fix this.

--
Rafael Carício

File: senza/cli.py [hunk 2]
```diff
@@ -1164,10 +1164,12 @@ def traffic(stack_name, stack_version, percentage, region, output):
                             break
                         elif current_stack_status.endswith('_IN_PROGRESS'):
                             # some operation in progress, let's wait some time to try again
+                            click.echo("Stack currently in state {}, waiting to perform traffic change...".format(
+                                current_stack_status))
                             time.sleep(5)
                         else:
-                            error("Stack is not in a valid state (%s), traffic change cannot be performed.",
-                                  current_stack_status)
+                            error("The traffic change cannot be performed in a stack in the {} state.".format(
```
Comment (hjacobs) [inline senza/cli.py]: nitpick: "The traffic change cannot be performed in a stack " -> "The traffic change cannot be performed on a stack "

**Comment by rafaelcaricio** [comment]
Done.

**Comment by hjacobs** [comment]
:+1:

**Comment by jmcs** [comment]
:+1:

**Pull request merged**